### ISSUE TITLE
Feat: Reproducible colour configuration

### ISSUE THREAD

**Issue by snaeil** [adriankarlen/textfox#72]
Currently it is necessary to install Firefox Color and manually select a theme, using the browser.
It would be great to be able to configure colours in a way this doesn't have to be done every time, textfox is installed on a new system.
I am not familiar enough with colours and couldn't find a way to configure Firefox Color.
Especially using nix home-manager it would be great to set colours directly in the configuration.
If you feel different and don't want colours to be controlled via textfox at all, you can close this issue of course.

**Comment by adriankarlen** [comment]
This ones a bit tricky, while I understand the idea behind having the colors set from within the theme I want the theme to be color agnostic as the main way that colors are distributed is usually by the Firefox color extension.

It would be nice to have a clearer way to define the colors in the css, and make them overridable via `config.css`. This way the nix flake could be used to set colors in addition. However a lot of the colors are simply not overriden, so this would require more overrides to be performed.

I guess one way would be to expose the firefox-color css-variables to the config.css. Would require minimal effort while increasing user configurability. This however feels quite half assed, I'll think some about it and do some testing.

**Comment by adriankarlen** [comment]
This feels like a nix-flake thing, so I am thinking of adding this option only to the nix-flake, since that is the only way that requires it's own syntax.

The way the `config.css` works is essentially solving this issue for any non nix-user as they can simply do whatever css they want in that file, or import any other css-file they want there.

**Comment by snaeil** [comment]
Can colours be configured by simply having FirefoxColor and setting css?
If yes this should be easy to do with nix and is there a documentation on css that can be set for FirefoxColor?

**Comment by adriankarlen** [comment]
> Can colours be configured by simply having FirefoxColor and setting css?
> If yes this should be easy to do with nix and is there a documentation on css that can be set for FirefoxColor?

Yes this is how the colors are set. Firefox Color defines a bunch of css variables that are then set throughout Firefox. I have not scoured through the docs personally though.

The way it is used in textfox is that some of these colors are set to accent, bg, border etc. My idea is that by exposing these colors in the nix module.

However seeing as the accent, bg and border already are overridable this might cause issues and makes the color configuration quite weird imo.

**Comment by nyukuru** [comment]
Fwiw, you can create your own extension defining colors through the [manifest.json](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/theme) and declaratively install that extension. The drawbacks is that it requires you to manually publish and get mozilla to sign each color configuration, this can be solved with some scripting and disabling requiring extension signing in preferences.

IMO I think it would be nice to expose options that map the same manifest.json keys to the respective css properties (completely avoiding any need for extra extensions, including Firefox color)

**Comment by adriankarlen** [comment]
I guess, but that is also some extensive work to add color configuration when firefox color is a some what standard way to manage themes in firefox colors and the most popular color schemes has a port for it eg: https://github.com/catppuccin/firefox, https://github.com/rose-pine/firefox.
